### PR TITLE
Extend get module artifacts API by download URL

### PR DESCRIPTION
**Pull request by herdt-michael**
Introduce request parameter to request download URLs when retrieving list of artifacts for a specific software module.

### CODE REVIEW

File: hawkbit-rest/hawkbit-mgmt-api/src/main/java/org/eclipse/hawkbit/mgmt/rest/api/MgmtRestConstants.java [hunk 1]
```diff
@@ -275,7 +275,7 @@ public final class MgmtRestConstants {
      * Request parameter if the artifact url handler should be used
      */
     public static final String REQUEST_PARAMETER_USE_ARTIFACT_URL_HANDLER = "useartifacturlhandler";
-
+    public static final String REQUEST_PARAMETER_WITH_DOWNLOAD_URL = "withdownloadurl";
```
Comment (stefbehl) [inline hawkbit-rest/hawkbit-mgmt-api/src/main/java/org/eclipse/hawkbit/mgmt/rest/api/MgmtRestConstants.java:278]: why do we need that? isn't it sufficient to append "?representation=full&useartifacturlhandler=true"?

File: hawkbit-rest/hawkbit-mgmt-resource/src/main/java/org/eclipse/hawkbit/mgmt/rest/resource/MgmtSoftwareModuleResource.java [hunk 2]
```diff
@@ -122,10 +123,19 @@ public ResponseEntity<MgmtArtifact> uploadArtifact(@PathVariable("softwareModule
 
     @Override
     public ResponseEntity<List<MgmtArtifact>> getArtifacts(
-            @PathVariable("softwareModuleId") final Long softwareModuleId) {
+            @PathVariable("softwareModuleId") final Long softwareModuleId,
+            @RequestParam(value = MgmtRestConstants.REQUEST_PARAMETER_WITH_DOWNLOAD_URL, required = false) final Boolean withDownloadUrl) {
```
Comment (stefbehl) [inline hawkbit-rest/hawkbit-mgmt-resource/src/main/java/org/eclipse/hawkbit/mgmt/rest/resource/MgmtSoftwareModuleResource.java:127]: couldn't we re-use the "representation mode" concept here? It is already used for other collections (e.g. for /rest/v1/actions or /rest/v1/rollouts)

File: hawkbit-rest/hawkbit-mgmt-api/src/main/java/org/eclipse/hawkbit/mgmt/rest/api/MgmtRestConstants.java (hunk 1 above)
Comment (herdt-michael) [reply hawkbit-rest/hawkbit-mgmt-api/src/main/java/org/eclipse/hawkbit/mgmt/rest/api/MgmtRestConstants.java]: Good point! I made of use the representation model and useartifacturlhandler parameter.

File: hawkbit-rest/hawkbit-mgmt-resource/src/main/java/org/eclipse/hawkbit/mgmt/rest/resource/MgmtSoftwareModuleResource.java (hunk 2 above)
Comment (herdt-michael) [reply hawkbit-rest/hawkbit-mgmt-resource/src/main/java/org/eclipse/hawkbit/mgmt/rest/resource/MgmtSoftwareModuleResource.java]: Good hint, changed

**Review by strailov — APPROVED** [review]

**Pull request merged**